hazelcast/src/main/java/com/hazelcast/config/IndexConfig.java

```diff
@@ -255,6 +257,9 @@ public void writeData(ObjectDataOutput out) throws IOException {
         out.writeInt(type.getId());
         writeNullableList(attributes, out);
         out.writeObject(bitmapIndexOptions);
+        if (out.getVersion().isGreaterOrEqual(Versions.V5_2)) {
```

Comment:
Think we need to add the below in both read and write. We look for these comments release by release and clean them (making the conditional serialization unconditional).
```
// RU_COMPAT 5.1
```